src/test_fixture/transport/channeled_transport.rs

```diff
@@ -0,0 +1,575 @@
+#![allow(dead_code)]
+
+use crate::helpers::query::QueryConfig;
+use crate::helpers::transport::{
+    ChannelledTransport, NoResourceIdentifier, QueryIdBinding, RouteId, RouteParams, StepBinding,
+};
+use crate::helpers::HelperIdentity;
+use crate::protocol::{QueryId, Step};
+use async_trait::async_trait;
+use futures::Stream;
+use futures::StreamExt;
+use futures_util::future::Either;
+use futures_util::stream;
+use serde::de::DeserializeOwned;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::fmt::{Debug, Formatter};
+use std::future::Future;
+use std::io;
+use std::pin::Pin;
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Poll, Waker};
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio_stream::wrappers::ReceiverStream;
+use tracing::Instrument;
+
+struct InMemoryPacket {
+    addr: RouteId,
+    origin: Option<HelperIdentity>,
+    query_id: Option<QueryId>,
+    step: Option<Step>,
+    params: String,
+}
+
+impl InMemoryPacket {
+    fn from_route<Q: QueryIdBinding, S: StepBinding, R: RouteParams<RouteId, Q, S>>(
+        origin: HelperIdentity,
+        route: &R,
+    ) -> Self
+    where
+        Option<QueryId>: From<Q>,
+        Option<Step>: From<S>,
+    {
+        Self {
+            addr: route.resource_identifier(),
+            origin: Some(origin),
+            query_id: route.query_id().into(),
+            step: route.step().into(),
```

Comment:
Given that we have two cases, would this be better as an enum:

```rust
#[derive(Debug)]
enum InMemoryPacket {
  Query, // This has nothing attached; RouteId::ReceiveQuery is implicit
  Data { // RouteId::Records is implicit
    origin: HelperIdentity,
    query: QueryId,
    step: Step,
  }
}
```

Reply:
I only added two cases for demonstration that this approach could work. We certainly have more, even now:

* receive query 
* prepare query
* receive inputs
* return query results